core/src/main/java/org/frankframework/receivers/Receiver.java

```diff
@@ -1119,24 +1124,39 @@ public void retryMessage(String storageKey) throws ListenerException {
 		}
 		try (PipeLineSession session = new PipeLineSession()) {
 			session.put(PipeLineSession.MANUAL_RETRY_KEY, true);
-			if (getErrorStorage()==null) {
+
+			PlatformTransactionManager txManager = getTxManager();
+			ITransactionalStorage<Serializable> errorStorage = getErrorStorage();
+			if (errorStorage == null) {
 				// if there is only a errorStorageBrowser, and no separate and transactional errorStorage,
 				// then the management of the errorStorage is left to the listener.
 				IMessageBrowser<?> errorStorageBrowser = messageBrowsers.get(ProcessState.ERROR);
-				RawMessageWrapper<?> msg = errorStorageBrowser.browseMessage(storageKey);
-				//noinspection unchecked
-				processRawMessage((RawMessageWrapper<M>) msg, session, true, false);
+				IbisTransaction itx = new IbisTransaction(txManager, getTxDef(), "receiver [" + getName() + "]");
+				try {
+					RawMessageWrapper<?> msg = errorStorageBrowser.browseMessage(storageKey);
+					//noinspection unchecked
+					processRawMessage((RawMessageWrapper<M>) msg, session, true, false);
+				} catch (ListenerException e) {
+					itx.setRollbackOnly();
+					throw e;
+				} catch (Throwable t) {
+					itx.setRollbackOnly();
+					throw new ListenerException(t);
+				} finally {
+					itx.complete();
+				}
 				return;
 			}
-			PlatformTransactionManager txManager = getTxManager();
-			IbisTransaction itx = new IbisTransaction(txManager, newTransaction, "receiver [" + getName() + "]");
 			RawMessageWrapper<Serializable> msg = null;
-			ITransactionalStorage<Serializable> errorStorage = getErrorStorage();
 			try {
+				IbisTransaction itx = new IbisTransaction(txManager, newTransaction, "receiver [" + getName() + "]");
```

Comment:
Ik vind hem technisch goed maar misschien dat het wel een goed idee is de doc aan te passen. bijv in `processRequest(..)` net als in `processRawMessage` iets van 'Creates a new transaction if necessary.' komt te staan, zodat duidelijk is welke methodes er een maken, en een verwachten (en om welke reden).

Ook denk ik dat `newTransaction` beter iets van `txRequiresNewWithTimeout` kan heten.